### ISSUE TITLE
Missing datetime format support 

### ISSUE THREAD

**Issue by epilys** [rusqlite/rusqlite#968]
According to https://sqlite.org/lang_datefunc.html the following string formats are expected by sqlite time functions:

1.  YYYY-MM-DD
2.  YYYY-MM-DD HH:MM
3.   YYYY-MM-DD HH:MM:SS
4.   YYYY-MM-DD HH:MM:SS.SSS
5.   YYYY-MM-DDTHH:MM
6.   YYYY-MM-DDTHH:MM:SS
7.   YYYY-MM-DDTHH:MM:SS.SSS
8.   HH:MM
9.    HH:MM:SS
10.   HH:MM:SS.SSS
11.    now
12.   DDDDDDDDDD 

rusqlite doesn't cover all of them, for example the fractional second part. That's `%Y-%m-%d %H:%M:%S %f` but `%f` is not supported by the `time` crate. It only has `%N` which is 9 digits. You can test if the string is 23 bytes long, and pad it with zeros:

```rust
if s.len() == 23 {
  let mut s = format!("{}000000", s);
}
```

Additionally, 

> Formats 2 through 10 may be optionally followed by a timezone indicator of the form "[+-]HH:MM" or just "Z". The date and time functions use UTC or "zulu" time internally, and so the "Z" suffix is a no-op

If the `Z` is missing, the string also doesn't parse correctly.

**Comment by gwenn** [comment]
Could you please be more specific ?
For example, you cannot reasonably transform an `HH:MM` into a `OffsetDateTime` !
And I don't expect `now` to be stored in an SQLite database.

**Comment by epilys** [comment]
Of course, as I said in the original post, a string like `2021-06-06 15:01:11.697` is valid for sqlite3 but cannot be parsed with rusqlite since the format strings here https://github.com/rusqlite/rusqlite/blob/840a6f3789f91d71599fa8fe909cc0d51b13f862/src/types/time.rs#L6-L8
don't recognize it.

Here are the formatting specifiers for the `time` crate: https://docs.rs/time/0.2.26/time/#formatting

The only subsecond specifier is

> %N | Subsecond nanoseconds. Always 9 digits

And the string  `2021-06-06 15:01:11.697` has 3 digits. The datetime strings in sqlite can recognize millisecond precision (3 digits) as stated in the link in the original post. So in order to parse it properly with %N, it needs to be padded with zeros. Furthermore, the format string with just `%N` has the `Z` character at the end but sqlite says it's optional, so the string `2021-06-06 15:01:11.697` cannot be validly parsed, though it should.

**Comment by gwenn** [comment]
https://github.com/time-rs/time/issues/329

**Comment by jhpratt** [comment]
Worth noting that time 0.3 has support for varying digits for the subsecond portion. There's an alpha release out now.

**Comment by epilys** [comment]
Here's a playground that parses all forms correctly:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f81dc7a6d02ff14cc969afe020df3cdf

@jhpratt  that's great :)

**Comment by epilys** [comment]
If you want I can make it a pull request, unless you want to do it another way

**Comment by gwenn** [comment]
Could you please do the same with optional offset / timezone [+-]HH:MM" ?
Or maybe we should drop support to `OffsetDateTime` (and support only `PrimitiveDateTime`) ?

**Comment by gwenn** [comment]
And these cases are still wrong:
```
2021-06-06 15:01:11.697Z, ("%Y-%m-%d %H:%M:%S.%NZ", Ok(PrimitiveDateTime { date: Date { year: 2021, ordinal: 157 }, time: Time { hour: 15, minute: 1, second: 11, nanosecond: 697 } }))
2021-06-06T15:01:11.697Z, ("%Y-%m-%dT%H:%M:%S.%NZ", Ok(PrimitiveDateTime { date: Date { year: 2021, ordinal: 157 }, time: Time { hour: 15, minute: 1, second: 11, nanosecond: 697 } }))
```

**Comment by gwenn** [comment]
Version 0.26.0 released
